nixos/modules/services/web-apps/etesync-web.nix

```diff
@@ -0,0 +1,50 @@
+{ lib, config, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.etesync-web;
+  etebaseHost = config.services.etebase-server.settings.allowed_hosts.allowed_host1;
+  pkg = pkgs.etesync-web.override { inherit (cfg) defaultServer; };
+  webRoot = "${pkg}/libexec/${pkg.pname}/deps/${pkg.pname}/build/";
+in
+{
+  meta.maintainers = with maintainers; [ pacman99 ];
+  options = {
+    services.etesync-web = {
+      enable = mkEnableOption "host etesync web interface with nginx";
+      hostName = mkOption {
+        type = types.str;
+        description = ''
+          The hostName to serve etesync-web from
+        '';
+      };
+      defaultServer = mkOption {
+        type = with types; nullOr str;
+        default = if etebaseHost != "0.0.0.0" then etebaseHost else null;
```

Comment:
Can you explain this line? Why is 0.0.0.0 not a valid value?

Reply:
I'll clarify with a comment. But the idea is that etebase-server's default allowed_host1 is 0.0.0.0, so I'm only setting the etebase host if its not the default.

Reply:
I think this is actually useful, since it won't cause `etesync-web` to be rebuilt for the default setup.

Reply:
I suggest `default = "0.0.0.0"`. The resulting package will be cached anyway due to the test.

Reply:
Thats not a working or sane default. If you enable etesync-web without an etebase server then the best default is the official etebase server instance(etesync.com). That way etesync-web will actually work without setting a server. 0.0.0.0 doesn't really mean anything.